cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/queue/AbstractQueueAdmin.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data2.transaction.queue;
+
+import co.cask.cdap.common.conf.CConfiguration;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.common.queue.QueueName;
+import co.cask.cdap.data2.datafabric.DefaultDatasetNamespace;
+import co.cask.cdap.data2.util.hbase.HBaseTableUtil;
+import co.cask.cdap.proto.Id;
+
+import static co.cask.cdap.data2.transaction.queue.QueueConstants.QueueType.QUEUE;
+
+/**
+ *
+ */
+public abstract class AbstractQueueAdmin implements QueueAdmin {
+  private final String unqualifiedTableNamePrefix;
+  protected final DefaultDatasetNamespace namespace;
+
+  public AbstractQueueAdmin(CConfiguration conf, QueueConstants.QueueType type) {
+    // system scope
+    unqualifiedTableNamePrefix = Constants.SYSTEM_NAMESPACE + "."
+      + (type == QUEUE ? QueueConstants.QUEUE_TABLE_PREFIX : QueueConstants.STREAM_TABLE_PREFIX);
+    namespace = new DefaultDatasetNamespace(conf);
+  }
+
+  /**
+   * @param queueTableName actual queue table name
+   * @return namespace id that this queue belongs to
+   */
+  public static String getNamespaceId(String queueTableName) {
+    // last three parts are namespaceId, appName and flow
+    String[] parts = queueTableName.split("\\.");
+    String namespaceId;
+    if (parts.length == 6) {
+      // cdap.<namespace>.system.queue.<app>.<flow>
+      namespaceId = parts[1];
+    } else {
+      throw new IllegalArgumentException(String.format("Unexpected format for queue table name. " +
+                                                         "Expected 'cdap.system.queue.<app>.<flow>' or " +
+                                                         "'cdap.system.queue.<namespace>.<app>.<flow>'. " +
```

Comment:
its just 6 now.
default is not being skipped here. That'll be handled by the logic that does hbase namespacing (since that sort of logic needs to be there anyways).